test.py

```diff
@@ -1099,6 +1099,33 @@ def test_fetch_malformed_url(self):
 
         self.assertEqual(expected_msg, str(cm.exception))
 
+    # FIXME: Won't work since file:// URLs are rejected
```

Comment:
This reminds me that we should have a test to confirm that `file://` raises an exception. I'm wondering whether that should also be the case except for a whitelisted set of schemes — `ftp`, `http`, and `https`? — with perhaps a way to add additional protocols.

Reply:
`file://` will work if it has a netloc, i.e.

```
file:///foo/bar
```

will raise an exception, but

```
file://server/foo/bar
```

will not.

It's probably wise to whitelist just HTTP protocols and provide a mechanism to extend that list (e.g. `s3://` for #117) at runtime.